src/internal/database/url-signing-jwk-generator.ts

```diff
@@ -0,0 +1,76 @@
+import { logger, logSchema } from '../monitoring'
+import { JwksCreateSigningSecret } from '@storage/events'
+import { listTenantsMissingUrlSigningJwk } from './tenant'
+import { getConfig } from '../../config'
+
+const { isMultitenant, pgQueueEnable } = getConfig()
+
+export interface UrlSigningJwkGeneratorStatus {
+  running: boolean
+  sent: number
+}
+
+export class UrlSigningJwkGenerator {
+  private static signal: AbortSignal
+  private static isRunning: boolean = false
+  private static countSent: number = 0
+
+  static Init(signal: AbortSignal) {
```

Comment:
this method it's not doing initialisation, maybe we can rename it to something like:

```
setAbortSignal()
```

Reply:
this shouldn't be needed, it's better to pass the signal in the method that can be aborted